src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -1961,16 +1961,26 @@ bool Compiler::optRedundantRelop(BasicBlock* const block)
             continue;
         }
 
-        // If the lcl defined here is live out, forward sub is problematic.
-        // We'll either create a redundant tree (as the original won't be dead)
-        // or lose the def (if we actually move the RHS tree).
+        // If the lcl defined here is live out we'll create a redundant tree
+        // (as the original won't be dead).
         //
         if (VarSetOps::IsMember(this, block->bbLiveOut, prevTreeLclDsc->lvVarIndex))
         {
             JITDUMP(" -- prev tree lcl V%02u is live-out\n", prevTreeLcl);
             continue;
         }
 
+        // The local may have uses between the def and the JTRUE. Duplicating
+        // reads is not allowed and in any case duplicating it in this case is
+        // probably not be worth it from a profitability perspective.
+        //
+        LclSsaVarDsc* ssaDefDsc = prevTreeLclDsc->GetPerSsaData(prevTreeLHS->AsLclVarCommon()->GetSsaNum());
+        if (ssaDefDsc->GetNumUses() >= 2)
```

Comment:
Notably, this it is not a 100% fix, since we do not count composite uses.

Reply:
Ah, well that's unfortunate as I was trying to simplify the substitution that happens below.
Do we see the composite case here?